api/document/tests/serializers/doc_cursor_test.py

```diff
@@ -295,6 +295,53 @@ def test_children_field_to_internal_value_works():
     assert doc_cursor.ChildrenField().to_internal_value([para]) == [para]
 
 
+def test_footnote_type_emblem_existence_is_validated():
+    serializer = doc_cursor.DocCursorSerializer()
+
+    footnote = f.footnote(1, [])
+
+    serializer.to_internal_value(footnote)
+
+    err_msg = "Footnotes must have type emblems"
+    with pytest.raises(ValidationError, match=err_msg):
+        del footnote['type_emblem']
+        serializer.to_internal_value(footnote)
+
+
+def test_footnote_type_emblem_uniqueness_is_validated():
+    serializer = doc_cursor.DocCursorSerializer()
+
+    serializer.to_internal_value(f.para([], children=[
+        f.footnote(1, []),
+        f.footnote(2, [])
+    ]))
+
+    err_msg = "Multiple footnotes exist with type emblem '1'"
+    with pytest.raises(ValidationError, match=err_msg):
+        serializer.to_internal_value(f.para([], children=[
+            f.para([], children=[f.footnote(1, [])]),
+            f.para([], children=[f.footnote(1, [])]),
```

Comment:
Any reason the second layer of `para`s is needed in this test? It's different from the "success" case, which makes me wonder if there's more going on.

Reply:
Yeah, the reason that there's two layers is because if the footnotes were the child of the same node, the "siblings are not unique across (node_type, type_emblem) values" validation would get triggered instead of the footnote one.